website/docs/guides/orchestration/custom-cicd-pipelines/1-creating-custom-cicd-pipelines.md

```diff
@@ -0,0 +1,431 @@
+    stage: pre-build
+    rules:
+    - if: $CI_PIPELINE_SOURCE == "push" && $CI_COMMIT_BRANCH != 'main'
+    script:
+    - pip install sqlfluff==0.13.1
+    - sqlfluff lint models --dialect snowflake --rules L019,L020,L021,L022
+```
+
+</details>
+
+
+
+### 2. Commit and push your changes to make sure everything works
+
+After you finish creating the yaml files, commit and push your code. Doing this will trigger your pipeline for the first time! If everything goes well, you should see the pipeline in your code platform. When you click into the job you’ll get a log showing that SQLFluff was run. If your code failed linting you’ll get an error in the job with a description of what needs to be fixed. If everything passed the lint check, you’ll see a successful job run.
+
+<details>
+<summary> GitHub </summary>
+
+In your repository, click the *Actions* tab
+
+![Image showing the GitHub action for lint on push](/img/guides/orchestration/custom-cicd-pipelines/lint-on-push-github.png)
+
+Sample output from SQLFluff in the `Run SQLFluff linter` job:
+
+![Image showing the logs in GitHub for the SQLFluff run](/img/guides/orchestration/custom-cicd-pipelines/lint-on-push-logs-github.png)
+</details>
+    
+<details>
+<summary> GitLab </summary>
+    
+In the menu option got to *CI/CD > Pipelines*
+
+![Image showing the GitLab action for lint on push](/img/guides/orchestration/custom-cicd-pipelines/lint-on-push-gitlab.png)
+
+Sample output from SQLFluff in the `Run SQLFluff linter` job:
+
+![Image showing the logs in GitLab for the SQLFluff run](/img/guides/orchestration/custom-cicd-pipelines/lint-on-push-logs-gitlab.png)
+</details>
+
+
+
+## Setting up dbt Cloud run on merge to main
+
+This job will take a bit more to setup, but is a good example of how to call the dbt Cloud API from a CI/CD pipeline. The concepts persented here can be generalized and used in whatever way best suits your use case.
+
+
+
+### 1. Get your dbt Cloud API key
+
+When running a CI/CD pipeline you’ll want to use a service token instead of any individual’s API key. There are [detailed docs](https://docs.getdbt.com/docs/dbt-cloud/dbt-cloud-api/service-tokens) available on this, but below is a quick rundown (this must be performed by an Account Admin):
+
+- Login to your dbt Cloud account
+- In the upper left, click the menu button, then *Account Settings*
+- Click *Service Tokens* on the left
+- Click *New Token* to create a new token specifically for CI/CD API calls
+- Name your token something like “CICD Token”
+- Click the *+Add* button under *Access,* and grant this token the *Job Admin* permission
+- Click *Save* and you’ll see a grey box appear with your token. Copy that and save it somewhere safe (this is a password, and should be treated as such).
+
+![View of the dbt Cloud page where service tokens are created](/img/guides/orchestration/custom-cicd-pipelines/dbt-service-token-page.png)
+
+Here’s a video showing the steps as well:
+
+<WistiaVideo id="iub17te9ir" />
+
```

Comment:
Can we remove this extra whitespace after the video?

Reply:
good call, will fix this